### ISSUE TITLE
Which loss should we monitor

### ISSUE THREAD

**Issue by XavierLinNow** [junyanz/pytorch-CycleGAN-and-pix2pix#31]
Hi, I used CycleGan to train new data. But I don't know which loss or metrics should I monitor. 
I wanted to choose loss G_A, but it seems to not to decrease. I also used the data horse2zebra, but loss G_A still didn't decrease a lot during 200 epochs.

**Comment by filmo** [comment]
Out of curiosity, how long did it take you do do 200 epochs? I'm currently training monet2photo and it's taking about 30 minutes per epoch using two GTX-1070s.

**Comment by XavierLinNow** [comment]
@filmo, training horse2zebra which contains 2400 pictures totally took about 1300 sec per epoch using TitanX. And the batch size is 1.

**Comment by filmo** [comment]
Thanks. monet2pic is 1831 seconds per epoch 6287 images.  2 x GTX-1070s. Batch size = 8 with identity set to 0.50.

Have you tried using a larger batch size?

Irrespective of the loss, did the end results look like they should have?

**Comment by XavierLinNow** [comment]
@filmo I have tried using batch size = 8. But the result is worse than batch size = 1 just based on my feeling. So I doubt if it has not converged. But there is no a good way to monitor it.

**Comment by filmo** [comment]
When using batch size = 8 did you use batch norm or instance norm?

**Comment by XavierLinNow** [comment]
@filmo, I used batch norm.

**Comment by taesung89** [comment]
Hi XavierLinNow, 

For GANs, monitoring the loss is not very informative, and I recommend looking at the generated images. If the resulting images are not satisfactory, try --use_dropout option, or --which_model_netD=n_layers with --n_layers_D=4
